### ISSUE TITLE
Update smithy-rs servers' dependencies on `http`, `http-body`, `hyper`, and `tower-http`.

### ISSUE THREAD

**Issue by david-perez** [smithy-lang/smithy-rs#3362]
Generated server SDKs use:

```
[dependencies.http]
version = "0.2.9"
[dev-dependencies.hyper]
version = "0.14.12"
```

And `aws-smithy-http-server` uses:

```
http = "0.2"
http-body = "0.4"
hyper = { version = "0.14.26", features = ["server", "http1", "http2", "tcp", "stream"] }
tower-http = { version = "0.3", features = ["add-extension", "map-response-body"] }
```

This is a breaking change because types from these crates are exposed in the `tower::Service` the generated SDKs export. `http`, `http-body`, `hyper` crates went 1.x.

Middleware authors using the 0.x versions of the `http` and `http-body` will be affected.

**Comment by cayman-amzn** [comment]
Looks like we might have this covered as apart of https://github.com/awslabs/aws-sdk-rust/issues/977

**Comment by djc** [comment]
Is this a duplicate of #1925?

**Comment by david-perez** [comment]
I think the only overlap of types depending on these crates used by both client and server is that servers ([unfortunately](https://github.com/smithy-lang/smithy-rs/issues/1649)) use `SdkBody`, but `SdkBody` has a [`from_body_1_x`](https://docs.rs/aws-smithy-types/latest/aws_smithy_types/body/struct.SdkBody.html#method.from_body_1_x), so we could give this a go server-side.

It'd be better if the upgrade in both server and client went out in the same smithy-rs release though, to minimize user disruption.

**Comment by nated0g** [comment]
the server sdks being on these older dependency versions is causing quite a bit of pain for us as we attempt to migrate a couple large rust services over from latest axum, which uses the newer http, hyper, etc, to the smithy-rs server sdks.

How much work would you estimate there is to do to get this closed out? You mention that it would be better for client/server to go out in the same release, but presumably the server stuff could be worked on separately and put up on a branch

Happy to submit a PR or something if you can point us in the right direction

**Comment by drganjoo** [comment]
Thanks for reaching out, and apologies for the delayed response. 

We are actively working on migrating the client runtime crates to Hyper 1, which should be released in the next few weeks. Following that, we plan to update the code generator to support Hyper 1 for both client and server-side implementations in a single release.

We understand this dependency mismatch is causing difficulties for teams working across different ecosystems, particularly where some crates (like Axum) have already migrated to Hyper 1 while smithy-rs hasn't yet. We're working to minimize this transition period. Once the runtime crates are updated, we expect the code generator updates to progress quickly.

We appreciate your patience and offer to help. We'll keep this issue updated with our progress.

**Comment by ZackKanter** [comment]
> We appreciate your patience and offer to help. We'll keep this issue updated with our progress.

Hi @drganjoo – just checking back to see if there was an update on this issue.

**Comment by ZackKanter** [comment]
> We plan on finishing this by the end of June 2025.

Hi @drganjoo – wanted to check in to see if this was still slated to land soon. Any update is greatly appreciated – we have some internal items that are blocked on this, so we're hoping to update our internal roadmap timing.

**Comment by drganjoo** [comment]
Apologies for the delay! Currently working on this and targeting completion in about three weeks.

**Comment by landonxjames** [comment]
We are working on this on the client side as well. You can see some of the initial work in https://github.com/smithy-lang/smithy-rs/pull/4181

**Comment by ZackKanter** [comment]
> Apologies for the delay! Currently working on this and targeting completion in about three weeks.

👋 @drganjoo – just checking to see if this was on track for shipping soon.

**Comment by cwoolum** [comment]
Any update on this?

**Comment by jlizen** [comment]
This should also sweep up #4242 as that is blocked on http 1x

**Comment by landonxjames** [comment]
This work was all covered in https://github.com/smithy-lang/smithy-rs/pull/4484 and will be releasing in the next few days

**Comment by jlizen** [comment]
@landonxjames has this shipped?

**Comment by drganjoo** [comment]
Yes this issue was shipped

**Comment by palfrey** [comment]
Except the `aws-smithy-runtime` with `test-util` feature still adds in `legacy-test-util`, which means hyper 0.x gets pulled in. See https://github.com/smithy-lang/smithy-rs/blob/8f0882046c467981ba5b4fd91a10befa96bd262f/rust-runtime/aws-smithy-runtime/Cargo.toml#L31

There is however a note above that about
```
# NOTE: `test-util` originally enabled the HTTP related test utils which have since been relocated to `aws-smithy-http-client`
# and updated to work with the 1.x HTTP/hyper ecosystem. Prefer using the `test-util` and `wire-mock` features from
# `aws-smithy-http-client` directly instead for HTTP test utils.
```
and using that directly solves this for our case.
